capa/features/extractors/base_extractor.py

```diff
@@ -467,4 +482,17 @@ def get_call_name(self, ph: ProcessHandle, th: ThreadHandle, ch: CallHandle) ->
         raise NotImplementedError()
 
 
+def ProcessFilter(extractor: DynamicFeatureExtractor, processes: Set) -> DynamicFeatureExtractor:
+    get_processes = extractor.get_processes  # fetch original get_functions()
+
+    def filtered_get_processes(self):
+        yield from (f for f in get_processes() if f.address.pid in processes)
+
+    # make a copy of the extractor before decorating the get_processes() method
+    new_extractor = copy(extractor)
+    new_extractor.get_processes = MethodType(filtered_get_processes, extractor)  # type: ignore
+
+    return new_extractor
```

Comment:
overall this approach seems pretty reasonable to me. as long as tests work, im happy!